### ISSUE TITLE
Remove heading rows downcast converter.

### ISSUE THREAD

**Issue by jodator** [ckeditor/ckeditor5#7601]
@scofalik & @Reinmar - I need to double check this as this changes how table is converted.

After numerous bugs with changing headings + removing rows we concluded that enough is enough. The table models [sucks](https://github.com/ckeditor/ckeditor5/issues/6506). Downcasting `headingRows` change requires moving `<tr>` in the view from `<thead>` to `<tbody>` or vice-versa. It can fail in multiple scenarios ([#6502](https://github.com/ckeditor/ckeditor5/issues/6502), [#6437](https://github.com/ckeditor/ckeditor5/issues/6437), [#6544](https://github.com/ckeditor/ckeditor5/issues/6544), [#6502](https://github.com/ckeditor/ckeditor5/issues/6502), [#6391](https://github.com/ckeditor/ckeditor5/issues/6391), [#6406](https://github.com/ckeditor/ckeditor5/issues/6406) + #7454) and would potentially never be OK.

Here (#7454) there was a problem with table layout post-fixer which was called after each `enqueueChange()` - and thus we hit limit of what API is allowing us to do. The solution is to wait with conversion and post-fixing after **all** table model changes are done and that `table` attribute converter _would know_ that also `table` children are changing.

Possible considered solutions were:

1.  A "hack": set heading rows to 0, remove/add rows, set heading rows to proper value - but it lacks an API for not converting stuff at once.
2.  Some API to mark that some changes need to be either converted at once or should not trigger model post-fixers.
3.  Mark `table` to be re-rendered at once.

We choose option `3` using `differ.refreshItem()` - which is called on every `headingRows` attribute operation as It requires to also take external operations into account when dealing with this bug. Also the API for many `TableUtils` get simplified but not passing a `batch` instance (:tada:).

---

In other words the solution is to re-render _whole_ table in the view if `headingRows` attribute was changed by any means.

---

What is needed - some kind of :+1: / :-1: - it might impact collaboration features in some way. Do we need better API for requiring re-converting whole chunk of content?

ps.: @Reinmar this is also some sort of ADR proposal (without sections - but have intro, considered steps and proposed solution).

_Originally posted by @jodator in https://github.com/ckeditor/ckeditor5/pull/7572_
